TokenCreation.sol

```diff
@@ -126,7 +126,7 @@ contract TokenCreation is TokenCreationInterface, Token {
                 extraBalance.payOut(address(this), extraBalance.accumulatedInput());
 
             // Execute refund
-            if (msg.sender.send(weiGiven[msg.sender])) { 
+            if (msg.sender.call.value(weiGiven[msg.sender])()) { 
```

Comment:
I have added `(balances[msg.sender] > 0 && weiGiven[msg.sender] > 0)` as pre-condition.

This is something like
```
    modifier onlyTokenholders {
        if (balanceOf(msg.sender) == 0) throw;
            _
    }
```
Before, I moved this modifier from `dao.sol` to `token.sol`, and added the modifier for the `refund()`, but the test would fail, so I changed it back.